internal/storage/postgres/tree/create.go

```diff
@@ -39,16 +39,28 @@ func (r *TreeRepository) Create(ctx context.Context, tFn ...entities.EntityFunc[
 }
 
 func (r *TreeRepository) CreateAndLinkImages(ctx context.Context, tFn ...entities.EntityFunc[entities.Tree]) (*entities.Tree, error) {
-	entity, err := r.Create(ctx, tFn...)
+	createdEntity, err := r.Create(ctx, tFn...)
 	if err != nil {
 		return nil, err
 	}
 
-	if err := r.handleImages(ctx, entity.ID, entity.Images); err != nil {
-		return nil, err
+	entity := defaultTree()
+	for _, fn := range tFn {
+		fn(&entity)
+	}
+
+	if entity.Images != nil {
```

Comment:
Why is this needed?

Reply:
Please clarify what you mean by "this".

If you are referring to the check entity.Images != nil, it's because the default struct initializes Images as nil. Imagine the function is called and Images is nil—what would happen in that case?